buildcontext/git.go

```diff
@@ -89,7 +89,7 @@ func (gr *gitResolver) resolveGitProject(ctx context.Context, target domain.Targ
 		return nil, "", "", fmt.Errorf("Invalid github project path %s", target.ProjectPath)
 	}
 	githubUsername := projectPathParts[0]
-	githubProject := projectPathParts[1]
+	githubProject := strings.TrimSuffix(projectPathParts[1], ".git")
```

Comment:
Hmm.. this should not be necessary? `.git` is not allowed in target references.

Reply:
I accidentally tried running this:
```
$ earth github.com/alexcb/hello.git+docker
buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)
ERROR: failed to fetch remote git@github.com:alexcb/hello.git.git: exit status 128
```
and got confused. I wonder if we should error out sooner saying the target is not valid.